 *******************************************************************************
 Copyright 2022-2023 Arm Limited and affiliates.
 Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 *******************************************************************************
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index 1261273bc92..201e00c6514 100644
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -198,9 +198,13 @@ def _tf_repositories():
         build_file = "//third_party/mkl_dnn:mkldnn_acl.BUILD",
         patch_file = [
             "//third_party/mkl_dnn:onednn_acl_threadcap.patch",
+            "//third_party/mkl_dnn:onednn_acl_remove_winograd.patch",
             "//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch",
             "//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch",
+            "//third_party/mkl_dnn:onednn_reorder_padded.patch",
+            "//third_party/mkl_dnn:onednn_reorder_update.patch",
             "//third_party/mkl_dnn:onednn_acl_threadpool_scheduler.patch",
+            "//third_party/mkl_dnn:onednn_acl_depthwise_convolution_dilation.patch",
         ],
         sha256 = "a50993aa6265b799b040fe745e0010502f9f7103cc53a9525d59646aef006633",
         strip_prefix = "oneDNN-2.7.3",
@@ -209,11 +213,10 @@ def _tf_repositories():

     tf_http_archive(
         name = "compute_library",
-        sha256 = "e20a060d3c4f803889d96c2f0b865004ba3ef4e228299a44339ea1c1ba827c85",
-        strip_prefix = "ComputeLibrary-22.11",
-        build_file = "//third_party/compute_library:BUILD",
-        patch_file = ["//third_party/compute_library:compute_library.patch", "//third_party/compute_library:acl_fixed_format_kernels_striding.patch", "//third_party/compute_library:acl_openmp_fix.patch"],
-        urls = tf_mirror_urls("https://github.com/ARM-software/ComputeLibrary/archive/v22.11.tar.gz"),
+        sha256 = "4c22983f08cbc26a7b66c695ee6850d39ea1346a6c76a902323dd10217df4606",
+        strip_prefix = "ComputeLibrary-23.05",
+        patch_file = ["//third_party/compute_library:compute_library.patch"],
+        urls = tf_mirror_urls("https://github.com/ARM-software/ComputeLibrary/archive/v23.05.tar.gz"),
     )

     tf_http_archive(
diff --git a/third_party/compute_library/BUILD b/third_party/compute_library/BUILD
index 68be8a22762..e69de29bb2d 100644
--- a/third_party/compute_library/BUILD
+++ b/third_party/compute_library/BUILD
@@ -1,189 +0,0 @@
-load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
-
-exports_files(["LICENSE"])
-
-cc_library(
-    name = "include",
-    hdrs = glob([
-        "include/**/*.h",
-        "include/**/*.hpp",
-    ]),
-    includes = ["include"],
-    strip_include_prefix = "include",
-)
-
-_COMPUTE_LIBRARY_DEFINES = [
-    "ARM_COMPUTE_OPENMP_SCHEDULER",
-    "ARM_COMPUTE_CPU_ENABLED",
-    "ENABLE_NEON",
-    "ARM_COMPUTE_ENABLE_NEON",
-    "ENABLE_SVE",
-    "ARM_COMPUTE_ENABLE_SVE",
-    "ARM_COMPUTE_ENABLE_BF16",
-    "ARM_COMPUTE_ENABLE_I8MM",
-    "ARM_COMPUTE_ENABLE_SVEF32MM",
-    "ENABLE_FP32_KERNELS",
-    "ENABLE_QASYMM8_KERNELS",
-    "ENABLE_QASYMM8_SIGNED_KERNELS",
-    "ENABLE_QSYMM16_KERNELS",
-    "ENABLE_INTEGER_KERNELS",
-    "ENABLE_NHWC_KERNELS",
-    "ENABLE_NCHW_KERNELS",
-    "ARM_COMPUTE_ENABLE_FIXED_FORMAT_KERNELS",
-]
-
-cc_library(
-    name = "arm_compute_sve2",
-    srcs = glob(
-        [
-            "src/cpu/kernels/**/sve2/*.cpp",
-            "**/*.h",
-            "**/*.hpp",
-            "**/*.inl",
-        ],
-    ),
-    copts = [
-        "-march=armv8.6-a+sve2",
-        "-fopenmp",
-    ],
-    defines = _COMPUTE_LIBRARY_DEFINES + ["ARM_COMPUTE_ENABLE_SVE2"],
-    includes = [
-        "src/core/NEON/kernels/arm_conv",
-        "src/core/NEON/kernels/arm_gemm",
-        "src/core/NEON/kernels/assembly",
-        "src/core/cpu/kernels/assembly",
-        "src/cpu/kernels/assembly",
-    ],
-    linkopts = ["-fopenmp"],
-    deps = ["include"],
-)
-
-cc_library(
-    name = "arm_compute_sve",
-    srcs = glob(
-        [
-            "src/core/NEON/kernels/arm_gemm/kernels/sve_*/*.cpp",
-            "src/core/NEON/kernels/arm_conv/**/kernels/sve_*/*.cpp",
-            "src/core/NEON/kernels/arm_conv/depthwise/interleaves/sve_*.cpp",
-            "src/core/NEON/kernels/batchnormalization/impl/SVE/*.cpp",
-            "src/core/NEON/kernels/convolution/winograd/input_transforms/sve_fp32_6x6.cpp",
-            "src/cpu/kernels/**/sve/*.cpp",
-            "**/*.h",
-            "**/*.hpp",
-            "**/*.inl",
-        ],
-    ) + [
-        "src/core/NEON/kernels/arm_gemm/transform-sve.cpp",
-        "src/core/NEON/kernels/arm_gemm/mergeresults-sve.cpp",
-    ],
-    copts = [
-        "-march=armv8.2-a+sve",
-        "-fopenmp",
-    ],
-    defines = _COMPUTE_LIBRARY_DEFINES,
-    includes = [
-        "src/core/NEON/kernels/arm_conv",
-        "src/core/NEON/kernels/arm_gemm",
-        "src/core/NEON/kernels/assembly",
-        "src/core/cpu/kernels/assembly",
-        "src/cpu/kernels/assembly",
-    ],
-    linkopts = ["-fopenmp"],
-    deps = ["include"],
-)
-
-cc_library(
-    name = "arm_compute",
-    srcs = glob(
-        [
-            "src/common/**/*.cpp",
-            "src/core/*.cpp",
-            "src/core/CPP/kernels/*.cpp",
-            "src/core/helpers/*.cpp",
-            "src/core/utils/**/*.cpp",
-            "src/runtime/**/*.cpp",
-            "src/c/*.cpp",
-            "src/core/NEON/kernels/*.cpp",
-            "src/core/NEON/kernels/convolution/**/*.cpp",
-            "src/core/NEON/kernels/arm_gemm/kernels/a64_*/*.cpp",
-            "src/core/NEON/kernels/arm_conv/pooling/*.cpp",
-            "src/core/NEON/kernels/arm_conv/**/kernels/a64_*/*.cpp",
-            "src/core/NEON/kernels/arm_conv/depthwise/*.cpp",
-            "src/core/NEON/kernels/arm_conv/depthwise/interleaves/a64_*.cpp",
-            "src/core/NEON/kernels/arm_conv/depthwise/interleaves/generic*.cpp",
-            "src/core/NEON/kernels/batchnormalization/impl/NEON/*.cpp",
-            "src/cpu/*.cpp",
-            "src/cpu/kernels/*.cpp",
-            "src/cpu/kernels/fuse_batch_normalization/**/*.cpp",
-            "src/cpu/kernels/*/generic/*.cpp",
-            "src/cpu/operators/**/*.cpp",
-            "src/cpu/utils/*.cpp",
-            "src/cpu/kernels/internal/*.cpp",
-            "src/cpu/kernels/**/neon/*.cpp",
-            "src/cpu/kernels/**/nchw/*.cpp",
-            "src/core/NEON/kernels/arm_gemm/*.cpp",
-            "**/*.h",
-            "**/*.hpp",
-            "**/*.inl",
-        ],
-        exclude = [
-            "src/core/utils/logging/**",
-            "src/core/TracePoint.cpp",
-            "src/core/NEON/kernels/arm_gemm/mergeresults-sve.cpp",
-            "src/core/NEON/kernels/arm_gemm/transform-sve.cpp",
-            "src/core/NEON/kernels/convolution/winograd/input_transforms/sve_fp32_6x6.cpp",
-            "src/runtime/CL/**",
-            "src/gpu/**",
-        ],
-    ) + [
-        "src/core/CPP/CPPTypes.cpp",
-        "src/c/operators/AclActivation.cpp",
-        "src/core/NEON/kernels/arm_conv/pooling/kernels/cpp_nhwc_1x1_stride_any_depthfirst/generic.cpp",
-        "src/core/NEON/kernels/arm_conv/depthwise/interleaves/8b_mla.cpp",
-        "src/core/NEON/kernels/arm_conv/addressing.cpp",
-    ],
-    hdrs = glob([
-        "src/core/NEON/kernels/**/*.h",
-        "src/core/NEON/kernels/**/*.hpp",
-        "arm_compute/runtime/**/*.h",
-        "arm_compute/runtime/*.h",
-        "arm_compute/core/**/*.h",
-        "**/*.inl",
-    ]) + [
-        "arm_compute_version.embed",
-    ],
-    copts = [
-        "-march=armv8-a",
-        "-fopenmp",
-    ],
-    defines = _COMPUTE_LIBRARY_DEFINES,
-    includes = [
-        "arm_compute/runtime",
-        "src/core/NEON/kernels/assembly",
-        "src/core/NEON/kernels/convolution/common",
-        "src/core/NEON/kernels/convolution/winograd",
-        "src/core/cpu/kernels/assembly",
-        "src/cpu/kernels/assembly",
-    ],
-    linkopts = ["-fopenmp"],
-    visibility = ["//visibility:public"],
-    deps = [
-        "arm_compute_sve",
-        "arm_compute_sve2",
-        "include",
-    ],
-)
-
-config_setting(
-    name = "build_with_acl",
-    define_values = {
-        "build_with_acl": "true",
-    },
-    visibility = ["//visibility:public"],
-)
-
-bzl_library(
-    name = "build_defs_bzl",
-    srcs = ["build_defs.bzl"],
-    visibility = ["//visibility:public"],
-)
diff --git a/third_party/compute_library/build_defs.bzl b/third_party/compute_library/build_defs.bzl
index 74102fd3e6d..3898798a42d 100644
--- a/third_party/compute_library/build_defs.bzl
+++ b/third_party/compute_library/build_defs.bzl
@@ -1,6 +1,6 @@
 def if_enable_acl(if_true, if_false = []):
     return select({
-        "@org_tensorflow//third_party/compute_library:build_with_acl": if_true,
+        "@compute_library//:build_with_acl": if_true,
         "//conditions:default": if_false,
     })
 
@@ -15,6 +15,6 @@ def acl_deps():
       inclusion in the deps attribute of rules.
     """
     return select({
-        "@org_tensorflow//third_party/compute_library:build_with_acl": ["@compute_library//:arm_compute"],
+        "@compute_library//:build_with_acl": ["@compute_library//:arm_compute_core"],
         "//conditions:default": [],
     })
diff --git a/third_party/mkl_dnn/mkldnn_acl.BUILD b/third_party/mkl_dnn/mkldnn_acl.BUILD
index a1085427ec0..cfbd515d781 100644
--- a/third_party/mkl_dnn/mkldnn_acl.BUILD
+++ b/third_party/mkl_dnn/mkldnn_acl.BUILD
@@ -173,6 +173,6 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        "@compute_library//:arm_compute",
+        "@compute_library//:arm_compute_core",
     ],
 )
